### PR TITLE
Memoise callbacks

### PR DESCRIPTION
**Pull request by ArthurWD**

### CODE REVIEW

File: src/components/Resource.tsx [hunk 1]
```diff
@@ -55,10 +55,11 @@ function useCalculatedViewWithState(props: InjectedPropTypes,
 export function LRC<P, R>(props: ResourcePropTypes<R> & P, _?: any): React.ReactElement | null {
     const context = useLinkRenderContext();
     const [error, setError] = React.useState<Error|undefined>(undefined);
+    const reset = React.useCallback(() => setError(undefined), []);
```
Comment (fletcher91) [inline src/components/Resource.tsx:58]: the dependency here is `setError`

File: src/hooks/useCalculateChildProps.ts [hunk 2]
```diff
@@ -33,6 +33,15 @@ export function useCalculateChildProps<P, R = any>(
...
     const overrides: Overrides<R> = {};
 
+    const reloadLinkedObject = React.useCallback(() => {
+      const toReload = props.subject || subject;
+      if (isNamedNode(toReload)) {
+        lrs.queueEntity(toReload, { reload: true });
+      }
+
+      return Promise.resolve();
+    }, [props.subject || subject]);
```
Comment (fletcher91) [inline src/hooks/useCalculateChildProps.ts:43]: this is missing `lrs`

File: src/components/Resource.tsx (hunk 1 above)
Comment (ArthurWD) [reply src/components/Resource.tsx:58]: setError will never change, because it comes from useState. Add it anyway?

Comment (fletcher91) [reply src/components/Resource.tsx:58]: Is it defined as not changing? Otherwise it could change on every render

Comment (ArthurWD) [reply src/components/Resource.tsx]: Could not find a definition for it. I added the dependencies anyway.

File: src/hooks/useCalculateChildProps.ts [hunk 3]
```diff
@@ -51,14 +60,7 @@ export function useCalculateChildProps<P, R = any>(
...
         overrides.reset = options.helpers.reset;
-        overrides.reloadLinkedObject = () => {
-          const toReload = props.subject || subject;
-          if (isNamedNode(toReload)) {
-            lrs.queueEntity(toReload, { reload: true });
-          }
-
-          return Promise.resolve();
-        };
+        overrides.reloadLinkedObject = reloadLinkedObject
```
Comment (fletcher91) [inline src/hooks/useCalculateChildProps.ts:63]: semicolon according to the CI

**Pull request merged**